### PR TITLE
Remove remaining functions from developer_tool

### LINKED ISSUE #159: Keystore path in roles_key_info should be expected to be relative to where the file is
The path is relative to where the command is called from and not to where the file is

### LINKED ISSUE #317: Remove remaining functionalities from `developer_tool`
Functions which are still in `developer_tool` all use `roles_key_infos`, expecting a `json` file containing information about the repository's roles. The question is if we want to keep using this `json` file. If this file is not provided the commands still work, but it's necessary to enter additional info. It makes sense to check if which data can we extracted from authentication repository's metadata files instead - unless this file is used to simplify commands like creation of a new repository and generation of new keystore file. When adding new roles, it should not be necessary to provide a file which contains information about all existing roles that should not be updated.
Once other functions using `roles_key_infos` are reworked, they should be moved to appropriate api modules. `developer_tool` shoud be removed.

- #159  can also be resolved as a part of cli refactroring and removal of `developer_tool`

### PR DESCRIPTION
**Pull request by renatav**
## Description (e.g. "Related to ...", etc.)

Closes #317 
Closes #159 

## Code review checklist (for code reviewer to complete)

- [ ] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR)
- [ ] Title summarizes what is changing
- [ ] Commit messages are meaningful (see [this][commit messages] for details)
- [ ] Tests have been included and/or updated, as appropriate
- [ ] Docstrings have been included and/or updated, as appropriate
- [ ] Changelog has been updated, as needed (see [CHANGELOG.md][changelog])

[changelog]: https://github.com/openlawlibrary/taf/blob/master/CHANGELOG.md
[commit messages]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Review by n-dusan — COMMENTED** [review]
Great work! Since we're removing `developer_tool.py` I think [this part](https://github.com/openlawlibrary/taf/blob/515c493e6e31b39a670ffa27f058bb043ca521e3/docs/developers/repository-classes.md?plain=1#L89-L97) of our documentation should also get updated?

Do we have documentation for taf commands with the new api? If we don't, that shouldn't hold up this PR. Usage examples would be helpful

How should we go about with #32 @renatav?

File: taf/api/roles.py [hunk 1]
```diff
@@ -95,6 +100,73 @@ def add_role_paths(
         auth_repo.commit(commit_message)
 
 
+def add_roles(
+    repo_path,
+    keystore=None,
+    roles_key_infos=None,
+    scheme=DEFAULT_RSA_SIGNATURE_SCHEME,
+):
```
[the remark this answers is not in the document]
Comment (renatav) [reply taf/api/roles.py:108]: Let's update all docstrings and documentation in a separate PR

[the remark this answers is not in the document]
Comment (n-dusan) [reply taf/api/roles.py:108]: Makes sense. I think we can add linting rules to our pre-commit for docstrings so we don't forget to add them.

**Review by n-dusan — APPROVED** [review]
We'll address updating documentation in a separate PR

**Pull request merged**